### PR TITLE
Add list of work(s) in which current work is nested on Dataset show p…

### PR DESCRIPTION
**Pull request by coblej**
…age.

### CODE REVIEW

File: app/indexers/dataset_indexer.rb [hunk 1]
```diff
@@ -13,7 +13,12 @@ class DatasetIndexer < Hyrax::WorkIndexer
   # Enable queries to distinguish top-level vs. nested datasets
   def generate_solr_document
    super.tap do |solr_doc|
-     solr_doc['top_level_bsi'] = object.in_works_ids.any? ? false : true
+     if object.in_works_ids.any?
+       solr_doc['top_level_bsi'] = false
+       solr_doc[Rdr::Index::Fields.in_works_ids] = object.in_works_ids
+     else
+       solr_doc['top_level_bsi'] = true
+     end
```
Comment (dchandekstark) [inline app/indexers/dataset_indexer.rb]: For the `if` block, consider:

```ruby
solr_doc['top_level_bsi'] = object.in_works_ids.blank? # or empty?
solr_doc[Rdr::Index::Fields.in_works_ids] = object.in_works_ids
```

File: app/indexers/dataset_indexer.rb [hunk 2]
```diff
@@ -13,7 +13,12 @@ class DatasetIndexer < Hyrax::WorkIndexer
   # Enable queries to distinguish top-level vs. nested datasets
   def generate_solr_document
    super.tap do |solr_doc|
-     solr_doc['top_level_bsi'] = object.in_works_ids.any? ? false : true
+     if object.in_works_ids.any?
+       solr_doc['top_level_bsi'] = false
```
Comment (dchandekstark) [inline app/indexers/dataset_indexer.rb]: We can create a field constant for this index field in `Rdr::Index::Fields`:

```ruby
register :top_level, nil, solr_name: 'top_level_bsi'
```

Comment (coblej) [reply app/indexers/dataset_indexer.rb]: @dchandekstark I don't disagree with this but think it's outside the scope of this PR.  Plus, there are some other Solr fields for which we should probably do the same thing, so doing all those together might be a good task / PR.

File: app/indexers/dataset_indexer.rb (hunk 1 above)
Comment (coblej) [reply app/indexers/dataset_indexer.rb]: @dchandekstark I do like this better.  PR revised accordingly.  (It did require a slight change in a spec test.)

**Pull request merged**